consensus/spos/bls/v2/subroundEndRound.go

```diff
@@ -93,127 +94,35 @@ func checkNewSubroundEndRoundParams(
 	return err
 }
 
-// receivedBlockHeaderFinalInfo method is called when a block header final info is received
-func (sr *subroundEndRound) receivedBlockHeaderFinalInfo(_ context.Context, cnsDta *consensus.Message) bool {
+func (sr *subroundEndRound) isProofForCurrentConsensus(proof consensus.ProofHandler) bool {
+	return bytes.Equal(sr.GetData(), proof.GetHeaderHash())
+}
+
+// receivedProof method is called when a block header final info is received
+func (sr *subroundEndRound) receivedProof(proof consensus.ProofHandler) {
 	sr.mutProcessingEndRound.Lock()
 	defer sr.mutProcessingEndRound.Unlock()
 
-	messageSender := string(cnsDta.PubKey)
-
+	if sr.IsJobDone(sr.SelfPubKey(), sr.Current()) {
```

Comment:
```suggestion
	if sr.IsSelfJobDone(sr.Current()) {
```

this one also covers multikey mode